File: Modules/Ticket/Priority/PriorityForm.cs
namespace Ticketing.Ticket.Forms;

[FormScript("Ticket.Priority")]
[BasedOnRow(typeof(PriorityRow), CheckNames = true)]
public class PriorityForm
{
    public string Name { get; set; }
    [ColorPickerEditor]
    public string Color { get; set; }
    public string Description { get; set; }
    public short DueTime { get; set; }
}

File: Modules/ServerTypes/Ticket/PriorityForm.ts
import { initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface PriorityForm {
    DueTime: IntegerEditor;
    Name: StringEditor;
    IconName: StringEditor;
    Color: StringEditor;
    Description: StringEditor;
}

export class PriorityForm extends PrefixedContext {
    static readonly formKey = 'Ticket.Priority';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PriorityForm.init) {
            PriorityForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;

            initFormType(PriorityForm, [
                'DueTime', w0,
                'Name', w1,
                'IconName', w1,
                'Color', w1,
                'Description', w1
            ]);
        }
    }
}